nwb_conversion_tools/json_schema_utils.py

```diff
@@ -59,9 +57,17 @@ def get_schema_from_method_signature(class_method, exclude=None):
     input_schema = get_base_schema()
     for param in inspect.signature(class_method).parameters.values():
         if param.name not in exclude + ['self']:
+            if param.name != 'self' and param.annotation:
+                if param.annotation is bool:
+                    param_type = "boolean"
+                if param.annotation is str:
+                    param_type = "string"
+            else:
+                raise NotImplementedError(f"The annotation type of '{param}' in function '{class_method}' "
+                                          "is not assigned! Please implement.")
```

Comment:
Thanks for fixing this! Would you mind refactoring this into a dictionary? Something like

```python
annotation_json_type_map = dict(
    bool="boolean",
    str="string",
    int="number",
    float="number"
)

Reply:
Yeah, I was thinking something like that might be the best way to go... Are those all the other 'simple' `json` schema types? Dictionaries would also be good to include, I'll check if that's supported

Reply:
[http://json-schema.org/understanding-json-schema/reference/type.html](http://json-schema.org/understanding-json-schema/reference/type.html) looks like it ought to